src/parser/GraphPatternOperation.cpp

```diff
@@ -59,19 +91,11 @@ void GraphPatternOperation::toString(std::ostringstream& os,
       // TODO<joka921> make the subquery a value-semantics type.
       os << arg.get().asString();
     } else if constexpr (std::is_same_v<T, Values>) {
-      os << "VALUES (";
-      for (const auto& v : arg._inlineValues._variables) {
-        os << v << ' ';
-      }
-      os << ") ";
-
-      for (const auto& v : arg._inlineValues._values) {
-        os << "(";
-        for (const auto& val : v) {
-          os << val << ' ';
-        }
-        os << ')';
-      }
+      // TODO: Why is there indentation in some of these strings, but not in
+      // others? And why do we have three different places that convert a SPARQL
+      // operation to a string?
+      os << "VALUES (" << arg._inlineValues.variablesToString() << ") "
+         << arg._inlineValues.valuesToString();
```

Comment:
This is only needed for some Unit Tests. 
It should be removed once we have better Matchers than "toString", 
so we just leave it here for now.

Reply:
(Answer for the TODO question)

Reply:
Thanks! I adapted my comment accordingly (and moved it to the definition of `GraphPatternOperation.cpp`, which so far was undocumented).